### PR TITLE
Embed SDK version

### PR DESCRIPTION
**Pull request by danielrbradley**
1. Change default config to use push-based Go SDK publishing
2. Use gopatch to enable RespectSchemaVersion in all languages
3. Rebuild SDKs if we've updated the resources.go file

Related to https://github.com/pulumi/ci-mgmt/issues/915 and https://github.com/pulumi/home/issues/3372

### CODE REVIEW

**Comment by iwahbe** [comment]
@danielrbradley Can you pick a bridged provider and deploy this code all the way through a release?

**Comment by danielrbradley** [comment]
Rolling out to pulumi-azure as a test:
- Action to create PR: https://github.com/pulumi/ci-mgmt/actions/runs/9059851553
- Azure PR: https://github.com/pulumi/pulumi-azure/pull/2028

**Comment by danielrbradley** [comment]
A couple of warts we might want to address that I spotted during the first test rollout:

- The SDK commit actually contains all the SDKs, not only the Go SDK. This is because we just point the Go SDK publish action as the module - which is `sdk/` - where the go.mod lives - which also includes all the other SDKs. This isn't a massive issue, but perhaps we could extend the action to allow globbing to select which files to include/exclude.
- Linked to the above issue, we also include the `go.tar.gz` file as we currently always download the assets into the sdk folder rather than a temp directory ... which results in it getting included when we publish the `sdk/...` go module.

**Comment by iwahbe** [comment]
> A couple of warts we might want to address that I spotted during the first test rollout:
> 
> * The SDK commit actually contains all the SDKs, not only the Go SDK. This is because we just point the Go SDK publish action as the module - which is `sdk/` - where the go.mod lives - which also includes all the other SDKs. This isn't a massive issue, but perhaps we could extend the action to allow globbing to select which files to include/exclude.
> * Linked to the above issue, we also include the `go.tar.gz` file as we currently always download the assets into the sdk folder rather than a temp directory ... which results in it getting included when we publish the `sdk/...` go module.

I think it's worth addressing both before we deploy to all providers. I don't want to regress in download size, so we **need** to address the second point before deploying. For simplicity sake, I would like to address the first point as well, but it's not necessary.

> but perhaps we could extend the action to allow globbing to select which files to include/exclude

Instead, can we just hard-code the list of paths we want to include? Right now, any pulumi provider should have all of their go code within `sdk/{go.mod,go.sum,go}`. `pulumi/publish-go-sdk-action@v1` can just know that.

**Review by iwahbe — COMMENTED** [review]
> A couple of warts we might want to address that I spotted during the first test rollout:
> 
> * The SDK commit actually contains all the SDKs, not only the Go SDK. This is because we just point the Go SDK publish action as the module - which is `sdk/` - where the go.mod lives - which also includes all the other SDKs. This isn't a massive issue, but perhaps we could extend the action to allow globbing to select which files to include/exclude.
> * Linked to the above issue, we also include the `go.tar.gz` file as we currently always download the assets into the sdk folder rather than a temp directory ... which results in it getting included when we publish the `sdk/...` go module.

I think it's worth addressing both before we deploy to all providers. I don't want to regress in download size, so we **need** to address the second point before deploying. For simplicity sake, I would like to address the first point as well, but it's not necessary.

> but perhaps we could extend the action to allow globbing to select which files to include/exclude

Instead, can we just hard-code the list of paths we want to include? Right now, any pulumi provider should have all of their go code within `sdk/{go.mod,go.sum,go}`. `pulumi/publish-go-sdk-action@v1` can just know that.

**Comment by danielrbradley** [comment]
> I think it's worth addressing both before we deploy to all providers. I don't want to regress in download size, so we **need** to address the second point before deploying. For simplicity sake, I would like to address the first point as well, but it's not necessary.

Done - make the action customisable with globs for the files to include, then customised the list of files for our use case within ci-mgmt as this might change in the future as we do things like move the go.mod one day or publish to a different repo.

File: tools/sourcemigrator/index.ts [hunk 1]
```diff
@@ -64,29 +61,65 @@ interface MigrateResult {
...
+          `go run github.com/uber-go/gopatch@latest -p "${patchPath}" ./provider/resources.go`
+        );
+        // Format the code - twice to ensure that the code is formatted correctly
+        run(`go install mvdan.cc/gofumpt@latest`);
+        run(`gofumpt -w ./provider/resources.go`);
+        run(`gofumpt -w ./provider/resources.go`);
+        // Check if we've made changes
+        const gitStatus = run(`git status --porcelain`).stdout;
+        if (gitStatus.includes("provider/resources.go")) {
+          run(`make tfgen build`);
```
Comment (VenelinMartinov) [inline tools/sourcemigrator/index.ts:93]: Do you not need to rebuild the sdks too?

Comment (danielrbradley) [reply tools/sourcemigrator/index.ts:93]: `make build` includes building the SDKs

**Pull request merged**